operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java

```diff
@@ -160,11 +158,22 @@ public Labels withAdditionalLabels(Map<String, String> additionalLabels) {
     }
 
     /**
-     * @param resource The resource to get the labels of
-     * @return the labels of the given {@code resource}.
+     * @param resource The resource to get the labels of.
+     * @return A new instance with filtered labels added from the given {@code resource}.
      */
     public static Labels fromResource(HasMetadata resource) {
-        return resource.getMetadata().getLabels() != null ? additionalLabels(resource.getMetadata().getLabels()) : EMPTY;
+        Map<String, String> additionalLabels = resource.getMetadata().getLabels();
+
+        if (additionalLabels != null) {
+            Pattern exclusionPattern = Pattern.compile(STRIMZI_LABELS_EXCLUSION_PATTERN);
```

Comment:
I think this would still compile the pattern anytime the method is called. The main idea is to have the compiled pattern as a static field itself. E.g. instead of:

```java
public static final String STRIMZI_LABELS_EXCLUSION_PATTERN = System.getenv()
            .getOrDefault("STRIMZI_LABELS_EXCLUSION_PATTERN", "^app.kubernetes.io/(?!part-of).*");
```

you do

```java
public static final Pattern STRIMZI_LABELS_EXCLUSION_PATTERN = Pattern.compile(System.getenv()
            .getOrDefault("STRIMZI_LABELS_EXCLUSION_PATTERN", "^app.kubernetes.io/(?!part-of).*"));
```

Reply:
got it, thanks! let me make the change and test it